### PR TITLE
Add mappings for `DisplayText` to `Text` component `README`

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris/issues/6670

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Text/README.md [hunk 1]
```diff
@@ -268,5 +268,145 @@ Use to de-emphasize a piece of text that is less important to merchants than oth
...
+#### Small
+
+```jsx
+<DisplayText size="small">Sales this year</DisplayText>
+```
+Replace with
+```jsx
+<Text variant="headingXl"
+>Sales this year</Text>
+```
```
Comment (alex-page) [inline polaris-react/src/components/Text/README.md]: This looks great Aveline. I was thinking about... what about using a [diff](https://stackoverflow.com/a/40883538)?

```suggestion
```diff
- <DisplayText size="small">Sales this year</DisplayText>
+ <Text variant="headingXl">Sales this year</Text>
```
```

File: polaris-react/src/components/Text/README.md [hunk 2]
```diff
@@ -268,5 +268,145 @@ Use to de-emphasize a piece of text that is less important to merchants than oth
   No supplier listed
```
Comment (alex-page) [inline polaris-react/src/components/Text/README.md:268]: If we want this to show up on the website, I would recommend moving this content to `polaris.shopify.com/content/components/text.md`

File: polaris-react/src/components/Text/README.md (hunk 1 above)
Comment (aveline) [reply polaris-react/src/components/Text/README.md]: Oh yes I like that, makes it more compact and clear. I will update them.

File: polaris-react/src/components/Text/README.md (hunk 2 above)
Comment (aveline) [reply polaris-react/src/components/Text/README.md:268]: So what content should go in this `README`?

Comment (alex-page) [reply polaris-react/src/components/Text/README.md:268]: Nothing. It's going to be deleted and I just migrated all the content to the site.

Comment (alex-page) [reply polaris-react/src/components/Text/README.md:268]: The examples are the final part to migrate.

Comment (aveline) [reply polaris-react/src/components/Text/README.md:268]: Ok got it! Thanks! 🧀

**Review by alex-page — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
lgtm! 🚀 💯 

just a thought, but should the other type components (`VisuallyHidden` and `TextContainer`) be included in the mapping too?

**Comment by aveline** [comment]
Good call @laurkim, I added `VisuallyHidden` but I think `TextContainer` will map to layout component?

**Comment by laurkim** [comment]
> Good call @laurkim, I added `VisuallyHidden` but I think `TextContainer` will map to layout component?

Oh yeah, that's a good point, I remember we had discussed that `<Stack>` replaces the same functionality 👍

**Review by sarahill — APPROVED** [review]

**Pull request merged**